### DISCUSSION TITLE
Does the tooltip need a close button?

### DISCUSSION THREAD

**Discussion by mikl** [ghostty-org/ghostty#3792] [Feature Requests, Ideas]
First of all, thanks for creating Ghostty, it’s a really nice piece of work.

I somehow managed to trigger a tooltip that covers the bottom left corner of the window. A bit inconvenient, when that’s also where the prompt is.

![Screenshot-2024-12-28-21 19 45](https://github.com/user-attachments/assets/288680e7-9bc1-4bf6-862c-2e6fe5e76076)

Not sure how I triggered this tooltip, I can’t figure out how to make it go away. I can mouse over it, which moves it to the bottom right instead, so I can see my prompt temporarily, but that’s not a permanent fix.

I realize this is probably user error, but it might be good with some sort of close button or another hint to the user how he can make the tooltip go away.

**Comment by mitchellh** [comment]
This should only show up if your mouse is hovering over a URL with a modifier pressed. If you can come up with reproduction steps so I can make this happen then I can try to improve the UX.

**Reply by mikl** [reply]
I managed to reproduce it (on macOS 15.2 (24C101) using Ghostty 1.0.0 build 8469) thusly:

1. Have a link in the terminal output (`echo "https://github.com/ghostty-org/ghostty/discussions/3792"` or whatever)
2. Hold down Cmd and hover over the link to make the it clickable.
3. Start dragging to select the link text.
4. Release Cmd while keeping the drag active. 
5. Move the mouse away from the link.
6. Tooltip is now stuck and doesn’t go away. 

This is also the way to clear the stuck tool-tip, simply Cmd-Hover another link without selecting anything and the tooltip goes away when the hover ends.

**Reply by 00-kat** [reply]
I can reproduce this on Linux as well.

> Move the mouse away from the link.

Just for extra clarification in case it helps, the selection should end outside the link itself, and it appears that ending it on top of another link also reproduces this bug.

**Comment by kellya** [comment]
Today I noticed that if the mouse happens to be over text in the terminal that it determines is a URL, and you hit some normal terminal combination like ctrl-r, or ctrl-c, ghostty is grabbing that ctrl, displaying the tool-tip, and sometimes getting stuck (though I couldn't reproduce the getting stuck part when I was trying to figure out what was going on, I have seen it).
